core/src/main.cpp

```diff
@@ -847,6 +849,22 @@ auto load_oidc_configuration(const json& _config, json& _oi_config, json& _endpo
 	return true;
 } // load_oidc_configuration
 
+auto load_user_mapping_plugin(const json& _user_mapping_config) -> void {
+	const auto& map_lib{_user_mapping_config["plugin_path"].get_ref<const std::string&>()};
+	const auto config_string{to_string(_user_mapping_config["configuration"])};
+
+	// Load shared lib
+	logging::trace("{}: Loading shared library at [{}]", __func__, map_lib);
+	boost::dll::shared_library user_map_lib{map_lib};
+	irods::http::globals::set_user_map_lib(user_map_lib);
+
+	// Get init func
+	auto init_func{user_map_lib.get<void (const char*)>("user_mapper_init")};
```

Comment:
I read that as, 'if name exists, regardless of specified signature, succeeds'.

The 'regardless of signature' part comes from me failing to update the signatures at some point, but still having the program compile/execute.

Are you suggesting we catch something here?

Reply:
Well, only if it throws. I'm saying that it is unclear from the docs whether that can happen (even the "**Throws:**" label is empty where every other function says "**Throws:** Nothing." if nothing is thrown). Basically, I'm wondering what happens "if call to `has(const char*) member function with the same symbol name returned" *false*.

I'm guessing that anything that is thrown will be caught... eventually... since there appears to be a try-catch block for `std::exception`s at the top level from which these things are being called.

Maybe it's moot for this case because we define the interface and the plugin is expected to have the required member function. Is this enforced in any way?

Reply:
Given that the plugin is only loaded on program start, using `.has()` seems good/ok. That will allow us to detect malformed plugins, log an error message, and end the program.

Reply:
So after shared library load (construction of the `user_map_lib` object), we then do a `.has(...)` check immediately to verify the symbol names can be found?

Reply:
Correct. If we can't find the symbols, log a error and terminate the program.

Reply:
Done.